### PR TITLE
[zephyr]making platform zephyr standalone

### PR DESCRIPTION
**Pull request by ck-telecom**
#### Problem
What is being fixed? 
making Zephyr as a standalone platform support

#### Change overview
add build and source support for zephyr standalone platform

#### Testing
How was this tested? (at least one bullet point required)
* If unit tests were added, how do they cover this issue?
* If unit tests existed, how were they fixed/modified to prevent this in future?
* If new unit tests are not added, why not?
* If integration tests were added, how do they verify this change?
* If new integration tests are not added, why not?
* If manually tested, what platforms controller and device platforms were manually tested, and how?
* If no testing is required, why not?

### CODE REVIEW

File: config/zephyr/.nrfconnect-recommended-revision [hunk 1]
```diff
@@ -0,0 +1 @@
+fd5905aa6b04febd99d00dba6c482ac25eb15222
```
Comment (Damian-Nordic) [inline config/zephyr/.nrfconnect-recommended-revision:1]: Not needed?

File: config/zephyr/README.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# nRF Connect build and configuration files
```
Comment (Damian-Nordic) [inline config/zephyr/README.md:1]: Needs be updated

File: config/zephyr/app/check-nrfconnect-version.cmake [hunk 3]
```diff
@@ -0,0 +1,28 @@
+#
```
Comment (Damian-Nordic) [inline config/zephyr/app/check-nrfconnect-version.cmake:1]: Needs be updated or removed for now

File: config/zephyr/chip-module/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+
+# TODO: fix hal nordic include path
+list(APPEND CHIP_CFLAGS -isystem${ZEPHYR_BASE}/../modules/hal/nordic/drivers/nrf_802154/driver/include)
+list(APPEND CHIP_CFLAGS -isystem${ZEPHYR_BASE}/../modules/hal/nordic/drivers/nrf_802154/sl/include)
+list(APPEND CHIP_CFLAGS -isystem${ZEPHYR_BASE}/../modules/crypto/mbedtls/include/)
+# Add MbedTLS config file
+#list(APPEND CHIP_CFLAGS -DMBEDTLS_CONFIG_FILE=CONFIG_MBEDTLS_CFG_FILE)
+
+# Add hardware cryptography dependencies
+#zephyr_include_directories(${ZEPHYR_BASE}/../modules/crypto/mbedtls/include)
```
Comment (Damian-Nordic) [inline config/zephyr/chip-module/CMakeLists.txt:132]: Remove commented lines?

**Review by Damian-Nordic — COMMENTED** [review]
I think it's worth cleaning some nRF Connect SDK bits at this stage.

File: config/zephyr/chip-module/CMakeLists.txt (hunk 4 above)
Comment (Damian-Nordic) [reply config/zephyr/chip-module/CMakeLists.txt:133]: Still not removed :)

File: config/zephyr/chip-module/Kconfig.defaults [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+	int
+	default 251
+
+config BT_RX_STACK_SIZE
+	int
+	default 1200
+
+# Enable NFC support
+# Latest Zephyr version V3.0 not supported
+# config CHIP_NFC_COMMISSIONING
```
Comment (Damian-Nordic) [inline config/zephyr/chip-module/Kconfig.defaults:175]: Remove?

File: config/zephyr/chip-module/Kconfig.defaults [hunk 6]
```diff
@@ -0,0 +1,240 @@
...
+
+config DEVMEM_SHELL
+	bool
+	default n
+
+config MCUBOOT_SHELL
+	bool
+	default n
+
+config CLOCK_CONTROL_NRF_SHELL
```
Comment (Damian-Nordic) [inline config/zephyr/chip-module/Kconfig.defaults:232]: Please remove

File: config/zephyr/chip-module/Kconfig.features [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+		Enables QSPI NOR with the set of options
+		configuring pages and buffer sizes.
+
+if CHIP_QSPI_NOR
+
+config NORDIC_QSPI_NOR
+	bool
+	default y
+
+config NORDIC_QSPI_NOR_STACK_WRITE_BUFFER_SIZE
```
Comment (Damian-Nordic) [inline config/zephyr/chip-module/Kconfig.features:34]: Please remove

File: config/zephyr/chip-module/Kconfig.mcuboot.defaults [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+
+config TICKLESS_KERNEL
+    bool
+    default n
+
+config TIMEOUT_64BIT
+    bool
+    default n
+
+config NRF_ENABLE_ICACHE
```
Comment (Damian-Nordic) [inline config/zephyr/chip-module/Kconfig.mcuboot.defaults:103]: Please remove

File: config/zephyr/chip-module/Kconfig.mcuboot.defaults [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+    bool
+    default n
+
+if SOC_SERIES_NRF53X
```
Comment (Damian-Nordic) [inline config/zephyr/chip-module/Kconfig.mcuboot.defaults:107]: Remove?

File: config/zephyr/chip-module/Kconfig.multiprotocol_rpmsg.defaults [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+
+config DEBUG_INFO
+    bool
+    default y
+
+config EXCEPTION_STACK_TRACE
+    bool
+    default y
+
+config NRF_802154_SER_RADIO
```
Comment (Damian-Nordic) [inline config/zephyr/chip-module/Kconfig.multiprotocol_rpmsg.defaults:93]: Remove nrf bits?

File: src/platform/Zephyr/BUILD.gn [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+
+    if (chip_mdns == "platform") {
+      sources += [ "../OpenThread/DnssdImpl.cpp" ]
+      deps = [ "${chip_root}/src/lib/dnssd:platform_header" ]
+    }
+  }
+
+  if (chip_enable_nfc) {
+    sources += [
+      "NFCManagerImpl.cpp",
```
Comment (Damian-Nordic) [inline src/platform/Zephyr/BUILD.gn:71]: Oh, looks like these two should be in `platform/nrfconnect` directory. We can do it as a followup though.

**Review by Damian-Nordic — APPROVED** [review]
Still found some pieces to remove, but looking good overall.

**Review by andy31415 — APPROVED** [review]

**Pull request merged**